pkg/aws/idps.go

```diff
@@ -34,72 +35,64 @@ const (
 		{
 			Key:   aws.String(tags.RedHatManaged),
 			Value: aws.String(tags.True),
 		},
 	}
 	if clusterID != "" {
-		iamTags = append(iamTags, &iam.Tag{
+		iamTags = append(iamTags, iamtypes.Tag{
 			Key:   aws.String(tags.ClusterID),
 			Value: aws.String(clusterID),
 		})
 	}
-	output, err := c.iamClient.CreateOpenIDConnectProvider(&iam.CreateOpenIDConnectProviderInput{
-		ClientIDList: []*string{
-			aws.String(OIDCClientIDOpenShift),
-			aws.String(OIDCClientIDSTSAWS),
+	output, err := c.iamClient.CreateOpenIDConnectProvider(context.TODO(), &iam.CreateOpenIDConnectProviderInput{
+		ClientIDList: []string{
+			OIDCClientIDOpenShift,
+			OIDCClientIDSTSAWS,
 		},
-		ThumbprintList: []*string{aws.String(thumbprint)},
-		Url:            aws.String(providerURL),
+		ThumbprintList: []string{thumbprint},
+		Url:            &providerURL,
 		Tags:           iamTags,
 	})
 	if err != nil {
 		return "", err
 	}
 
-	return aws.StringValue(output.OpenIDConnectProviderArn), nil
+	return aws.ToString(output.OpenIDConnectProviderArn), nil
 }
 
-func (c *awsClient) HasOpenIDConnectProvider(issuerURL string, accountID string) (bool, error) {
+func (c *awsClient) HasOpenIDConnectProvider(issuerURL string, partition string, accountID string) (bool, error) {
 	parsedIssuerURL, err := url.ParseRequestURI(issuerURL)
 	if err != nil {
 		return false, err
 	}
 	providerURL := fmt.Sprintf("%s%s", parsedIssuerURL.Host, parsedIssuerURL.Path)
 
-	oidcProviderARN := GetOIDCProviderARN(accountID, providerURL)
-	output, err := c.iamClient.GetOpenIDConnectProvider(&iam.GetOpenIDConnectProviderInput{
+	oidcProviderARN := GetOIDCProviderARN(partition, accountID, providerURL)
+	output, err := c.iamClient.GetOpenIDConnectProvider(context.TODO(), &iam.GetOpenIDConnectProviderInput{
 		OpenIDConnectProviderArn: aws.String(oidcProviderARN),
 	})
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			switch aerr.Code() {
-			case iam.ErrCodeNoSuchEntityException:
-				return false, nil
-			default:
-				return false, err
-			}
+		if awserr.IsNoSuchEntityException(err) {
+			return false, nil
 		}
+		return false, err
 	}
-	if aws.StringValue(output.Url) != providerURL {
+	if aws.ToString(output.Url) != providerURL {
 		return false, fmt.Errorf("The OIDC provider exists but is misconfigured")
 	}
 	return true, nil
 }
 
 func (c *awsClient) DeleteOpenIDConnectProvider(oidcProviderARN string) error {
-	_, err := c.iamClient.DeleteOpenIDConnectProvider(&iam.DeleteOpenIDConnectProviderInput{
+	_, err := c.iamClient.DeleteOpenIDConnectProvider(context.TODO(), &iam.DeleteOpenIDConnectProviderInput{
 		OpenIDConnectProviderArn: aws.String(oidcProviderARN),
 	})
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			switch aerr.Code() {
-			case iam.ErrCodeNoSuchEntityException:
-				return fmt.Errorf("OIDC provider '%s' does not exists",
-					oidcProviderARN)
-			}
+		if awserr.IsNoSuchEntityException(err) {
+			return fmt.Errorf("The OIDC provider does not exist")
```

Comment:
Should we include the OIDC provider ARN as we had previously in the error message?

Reply:
+1 to including the `arn` in the error message as before.